packages/pubsub/src/connection-pool.js

```diff
@@ -142,16 +146,25 @@ ConnectionPool.prototype.acquire = function(id, callback) {
  * @param {?error} callback.error - An error returned while closing the pool.
  */
 ConnectionPool.prototype.close = function(callback) {
-  var self = this;
   var connections = Array.from(this.connections.values());
 
   callback = callback || common.util.noop;
 
-  this.isOpen = false;
-  self.connections.clear();
-  this.queue.forEach(clearTimeout);
+  if (this.client) {
+    this.client.close();
+  }
 
+  this.connections.clear();
+  this.queue.forEach(clearTimeout);
   this.queue.length = 0;
+
+  this.isOpen = false;
+  this.waiting = false;
+
+  this.removeAllListeners('newListener')
```

Comment:
Are we the only ones who access `this`? Just thinking this could be bad if users might also have assigned a listener to `newListener` in their own apps for their own reasons.

Reply:
We should be, yes, the pool isn't even documented.